kratos/python_scripts/orchestrators/orchestrator.py

```diff
@@ -0,0 +1,186 @@
+            analysis_stage_module_name = input_analysis_stage
+            analysis_stage_class_name = analysis_stage_module_name.split('.')[-1]
+            analysis_stage_class_name = ''.join(x.title() for x in analysis_stage_class_name.split('_'))
+
+        # Import the stage module
+        # Note that for the case of custom analysis stage (i.e. no Kratos module) we check if the class exists in the __main__ module
+        if analysis_stage_module_name is not None:
+            analysis_stage_module = importlib.import_module(analysis_stage_module_name)
+        else:
+            analysis_stage_module = sys.modules["__main__"]
+
+        # Create the analysis stage instance
+        input_stage_settings = KratosMultiphysics.Parameters(current_stage_settings)
+        if hasattr(analysis_stage_module, analysis_stage_class_name):
+            # First we check for the expected class name
+            analysis_stage_class = getattr(analysis_stage_module, analysis_stage_class_name)
+            stage_instance = analysis_stage_class(self.__project.GetModel(),  input_stage_settings)
+        elif hasattr(analysis_stage_module, "Create"):
+            # If Kratos convention is not fulfilled we search for a Create method
+            stage_instance = analysis_stage_module.Create(self.__project.GetModel(), input_stage_settings)
+        else:
+            err_msg = f"Analysis stage in '{analysis_stage_module_name}' Python module cannot be created. Please check class name or provide a 'Create' method."
+            raise Exception(err_msg)
+
+        return stage_instance
+
+    def RunCurrentStagePreprocess(self, stage_name: str, data: typing.Optional[dict] = None):
+        """This function executes the preprocess of current stage.
+
+        Note that the stage preprocess involves the execution of modelers and operations.
+
+        Keyword arguments:
+        data -- Custom data argument to be used in derived classes
+        """
+
+        if self.__project.GetSettings()["stages"][stage_name].Has("stage_preprocess"):
+            if self.__project.GetSettings()["stages"][stage_name]["stage_preprocess"].Has("modelers"):
+                modelers_list = self.__CreateListOfModelers(stage_name)
+                for modeler in modelers_list:
+                    modeler.SetupGeometryModel()
+                for modeler in modelers_list:
+                    modeler.PrepareGeometryModel()
+                for modeler in modelers_list:
+                    modeler.SetupModelPart()
+                del modelers_list
+
+            if self.__project.GetSettings()["stages"][stage_name]["stage_preprocess"].Has("operations"):
+                operations_list = self.__CreateListOfOperations("stage_preprocess", stage_name)
+                for operation in operations_list:
+                    operation.Execute()
+                del operations_list
+
+    def RunCurrentStagePostprocess(self, stage_name: str, data: typing.Optional[dict] = None):
```

Comment:
I would do `from typing import Dict, Optional` at the top

Reply:
Done. Both in this file and in `project.py`.